datamade/ccfp-asset-dashboard
# Added a filterable data table + a JSON list view for the table

## Overview
This PR improves the existing list view for the projects. 

Added:
- filtering by project category and project section
- searching by project name and project description
- sorting by column
- a json list view for all the projects
- table pagination

Closes #26.

### Demo
![project-list-demo](https://user-images.githubusercontent.com/38969506/106063348-a8158480-60bd-11eb-9196-a7d076df787d.gif)


### Notes
There is some basic styling and layout but this PR still doesn't fit the CCFP's style guide.

## Testing Instructions
- Visit https://ccfp-asset-d-project-li-qeexkx.herokuapp.com/projects/
- Add some projects. At least 11 so you can test the pagination.
- Test all of the filtering/searching.
- Test the column sorting.
- Make sure you can click on a row to visit the project's detail page.
